File: CS50/Week 1/Problem Set 1/cash/cash.c
#include <cs50.h>
#include <stdio.h>

// Initialize constants

#define QUARTER 25
#define DIME 10
#define NICKEL 5
#define PENNY 1

int calculate_coins_used(int change);

int main(void)
{
    int change_owed;

    // Keep prompting the user to enter valid change
    do
    {
        change_owed = get_int("Change owed: ");
    }
    while (change_owed < 0);

    // Print the amount of coins used to give change
    printf("%i\n", calculate_coins_used(change_owed));
}

// Function to calculate the amount of coins used to give change
int calculate_coins_used(int change)
{
    // Initialize variables
    int quarter_count = 0;
    int dime_count = 0;
    int nickel_count = 0;
    int penny_count = 0;

    /*
        The while loop will calculate the amount of coins used to give change
        Will keep looping till the change amount is equal to 0
    */
    while (change > 0)
    {
        if (change >= QUARTER)
        {
            change = change - QUARTER;
            quarter_count++;
        }
        else if (change >= DIME)
        {
            change = change - DIME;
            dime_count++;
        }
        else if (change >= NICKEL)
        {
            change = change - NICKEL;
            nickel_count++;
        }
        else if (change >= PENNY)
        {
            change = change - PENNY;
            penny_count++;
        }
    }

    // Calculate and return total amount of coins used to give change
    int total = quarter_count + dime_count + nickel_count + penny_count;
    return total;
}

File: CS50/Week 6/Problem Set 6/sentimental-cash/cash.py
# Function to calculate the amount of coins used to give change
def calculate_coins_used(change):
    # Initialize constants
    QUARTER = 25
    DIME = 10
    NICKEL = 5
    PENNY = 1

    # Initialize variables
    quarter_count = 0
    dime_count = 0
    nickel_count = 0
    penny_count = 0

    # The while loop will calculate the amount of coins used to give change
    # Will keep looping till the change amount is equal to 0
    while change > 0:
        if change >= QUARTER:
            change = change - QUARTER
            quarter_count += 1
        elif change >= DIME:
            change = change - DIME
            dime_count += 1
        elif change >= NICKEL:
            change = change - NICKEL
            nickel_count += 1
        elif change >= PENNY:
            change = change - PENNY
            penny_count += 1

    # Calculate and return total amount of coins used to give change
    total = quarter_count + dime_count + nickel_count + penny_count

    return total


def main():
    change_owed = 0

    # Keep prompting the user to enter valid change
    while True:
        try:
            change_owed = int(float(input("Change owed: ")) * 100)
            if change_owed > 0:
                break
        except ValueError:
            pass

    # Print the amount of coins used to give change
    print(f"{calculate_coins_used(change_owed)}")


if __name__ == "__main__":
    main()
